src/main/java/com/avispl/symphony/dal/communicator/barco/e2gen2/BarcoE2Communicator.java

```diff
@@ -0,0 +1,999 @@
+		}
+		initializeData(statistics, controls);
+		localExtendedStatistics.setStatistics(statistics);
+		localExtendedStatistics.setControllableProperties(controls);
+		return Collections.singletonList(localExtendedStatistics);
+	}
+
+	/**
+	 * Call post request on the device to get RpcResponse DTO
+	 *
+	 * @param method String name of the method
+	 * @param param Map of params
+	 * @return JsonNode returns the JsonNode of given object
+	 * @throws Exception Throw exception when failed to call post request, get data from device
+	 */
+	protected JsonNode getByMethod(String method, Map<Object, Object> param) throws Exception {
+		RpcResponse rpcResponse;
+		try {
+			rpcResponse = this.doPost(BarcoE2Constant.DOUBLE_QUOTES, rpcRequestBody(method, param), RpcResponse.class);
+			if (rpcResponse == null || rpcResponse.getSuccessCode() != 0) {
+				throw new ResourceNotReachableException("doPost success but failed to get data from the device");
+			}
+			JsonNode response = rpcResponse.getResponse();
+			if (response != null && !NullNode.instance.equals(response) && !response.isEmpty()) {
+				return response;
+			}
+			return objectMapper.valueToTree(rpcResponse);
+		} catch (Exception e) {
+			logger.error(String.format("Failed to doPost for method %s", method));
+			throw e;
+		}
+	}
+
+	/**
+	 * Initialize monitoring/controlling data
+	 *
+	 * @param statistics Map of statistics
+	 * @param controls List of AdvancedControllableProperty that need to be controlled
+	 * @throws Exception Throw exception when failed to get preset/routing monitoring and controlling properties
+	 */
+	private void initializeData(Map<String, String> statistics, List<AdvancedControllableProperty> controls) throws Exception {
+		if (isFirstTimeMonitor) {
+			getDeviceInformation(statistics);
+			isFirstTimeMonitor = false;
+		}
+		getPresetFeedBack(statistics, controls);
+		getRoutingControl(true, statistics, controls);
+		getRoutingControl(false, statistics, controls);
+		getSuperRoutingControl(false, statistics, controls);
+		// TODO: routing monitoring/controlling for super destination
+	}
+
+	/**
+	 * Generate RpcRequest DTO
+	 *
+	 * @param method Rpc method
+	 * @param params Map of parameters
+	 * @return This returns RpcRequest DTO
+	 */
+	private RpcRequest rpcRequestBody(String method, Map<Object, Object> params) {
+		RpcRequest rpcRequest = new RpcRequest();
+		rpcRequest.setMethod(method);
+		rpcRequest.setParams(params);
+		return rpcRequest;
+	}
+
+	/**
+	 * Map a JsonNode to DTO
+	 *
+	 * @param jsonNode input jsonNode that need to be converted
+	 * @param tClass Class that will be converted to
+	 * @return This returns the DTO of given Object.
+	 */
+	private <T> Object jsonNodeToDTO(JsonNode jsonNode, Class<T> tClass) throws JsonProcessingException {
+		try {
+			return objectMapper.treeToValue(jsonNode, tClass);
+		} catch (JsonProcessingException e) {
+			logger.error("Failed to convert jsonNode to DTO");
```

Comment:
is it necessary to catch/rethrow and log?

Reply:
Updated. I delete the catch exception here.